execution_engine_testing/test_support/src/chainspec_config.rs

```diff
@@ -0,0 +1,194 @@
+use std::{
+    convert::TryFrom,
+    fs, io,
+    path::{Path, PathBuf},
+};
+
+use num_rational::Ratio;
+use once_cell::sync::Lazy;
+use serde::{Deserialize, Serialize};
+
+use casper_execution_engine::{
+    core::engine_state::{run_genesis_request::RunGenesisRequest, ExecConfig, GenesisAccount},
+    shared::{system_config::SystemConfig, wasm_config::WasmConfig},
+};
+use casper_types::ProtocolVersion;
+
+use crate::{
+    DEFAULT_ACCOUNTS, DEFAULT_CHAINSPEC_REGISTRY, DEFAULT_GENESIS_CONFIG_HASH,
+    DEFAULT_GENESIS_TIMESTAMP_MILLIS,
+};
+
+/// The name of the chainspec file on disk.
+pub const CHAINSPEC_NAME: &str = "chainspec.toml";
+
+/// Path to the production chainspec used in the Casper mainnet.
+pub static PRODUCTION_PATH: Lazy<PathBuf> = Lazy::new(|| {
+    PathBuf::from(env!("CARGO_MANIFEST_DIR"))
+        .join("resources/")
+        .join(CHAINSPEC_NAME)
+});
+
+#[derive(Debug)]
+pub enum Error {
+    FailedToLoadChainspec {
+        /// Path that failed to be read.
+        path: PathBuf,
+        /// The underlying OS error.
+        error: io::Error,
+    },
+    FailedToParseChainspec(toml::de::Error),
+    CouldNotCreateExecConfig,
+    CouldNotParseLockedFundsPeriod,
+    CouldNotCreateGenesisRequest,
```

Comment:
`FailedTo` rather than `CouldNot` for consistency?

Reply:
Renamed to maintain consistency, but introduced a clippy exception